collector/generic_collector.go

```diff
@@ -45,8 +57,20 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 
 	//TODO : Add checks for validity of config file (eg : Accurate JSON fields)
 
+	minPollFrequency := configInJSON.MetricsConfig[0].PollingFrequency
+	regexprs := make([]*regexp.Regexp, len(configInJSON.MetricsConfig))
+
+	for ind, metricConfig := range configInJSON.MetricsConfig {
+		if metricConfig.PollingFrequency < minPollFrequency {
+			minPollFrequency = metricConfig.PollingFrequency
+		}
+
+		//regexprs[ind] is set to nil if metricConfig.Regex cannot be parsed
+		regexprs[ind], _ = regexp.Compile(metricConfig.Regex)
```

Comment:
My thought (and I think Rohit's too), is that here we're able to tell the user that their config is wrong and allow them to fix it.

You bring up a good point about trying to fail with reduced functionality (but in that case, we should at least return the error). Only concern is that if some metrics show up and some don't the user would think that something was broken in cAdvisor (or that the metric is empty), whereas a clear yes no to the metric would possibly make them think the config was wrong.

We can go either way :) but I think this makes it clear that we should be doing very good reporting of errors on metric collection so users know what's happening. We should export some stats and possibly a page.